### ISSUE TITLE
Merge sample/api folders for Korro

### ISSUE THREAD

**Issue by Jolanrensen** [Kotlin/dataframe#898]
Code samples for the documentation website reside in [core/.../test/.../samples/api](./core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api),
and [tests/.../samples/api](./tests/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api)
and they are copied over to Markdown files in [docs/StardustDocs/topics](./docs/StardustDocs/topics)
by [Korro](https://github.com/devcrocod/korro).

What I do not know is why we have two separate folders for these samples. It's a bit confusing, so unless we have a reason not to, we should probably merge the two.

**Comment by AndreiKingsley** [comment]
So. I tried different things. 
1) "tests" module can no be moved into the "core/test" - it depends on Kandy which requires jvm >=11.
2) "samples" from "core/test" are really hard to move to the "tests" module - there're a lot of internal staff.

So my solution is simple.
1) Keep "tests" module (maybe rename it to "samples") and use it for creating DF and Kandy plot samples for docs.
2) After migration to new HTML dataframes in docs, we can remove old "samples" tests (except for those used by Kodex in KDocs.).

**Comment by Jolanrensen** [comment]
> "tests" module can no be moved into the "core/test" - it depends on Kandy which requires jvm >=11.

Agree, not only that, but it depends on other modules too.

> "samples" from "core/test" are really hard to move to the "tests" module - there're a lot of internal stuff.

Friend module? :))

But yes, sounds like a plan!

**Comment by AndreiKingsley** [comment]
> Friend module? :))

I tried, but there're a lot of issues with cross-dependencies, so it would take too much time to resolve all of them.

**Comment by AndreiKingsley** [comment]
I would reformulate what needs to be done within this issue as follows:

1) Remove from `:core` all code samples that are inserted into Writerside sources using Korro.
    * we can then make special samples for KDocs in future
2) Deprecate `DFExplainer` in favor of `SampleHelper` and rewrite all existing samples with SampleHelper.

`:samples` (former `:tests`) module now serves as unified WS samples source and kandy-plot/df-iframe samples handler.
`:core` tests should not contain WS samples (but might be for Kdocs)

@Jolanrensen do you agree?

**Comment by Jolanrensen** [comment]
@AndreiKingsley yes, that sounds like a good plan! Hopefully using `SampleHelper` we can drop the need for internal access. This was one of the things holding us back from extracting the samples from :core

**Comment by zaleslaw** [comment]
Moving this to 1.0 as a documentation infrastructure enabler.

This is not directly user-facing, but it supports the current 1.0 priority around documentation and examples. Writerside/Korro samples should not live in `:core` as pseudo-tests, and migrating them to `:samples` / `SampleHelper` should make documentation examples easier to maintain and validate during the 1.0 preparation.

Scope should stay focused: remove only Writerside/Korro documentation samples from `:core`, keep true library tests and possible KDoc-specific samples separate, and avoid unrelated sample-infra refactoring.
